src/Packages/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/build/Microsoft.CodeDom.Providers.DotNetCompilerPlatform.targets

```
@@ -9,9 +9,7 @@
   <Target Name="SetRoslynCompilerFiles" >
     <Message Text="Using Roslyn from '$(RoslynToolPath)' folder" />
     <ItemGroup>
-      <RoslynCompilerFiles Include="$(RoslynToolPath)\*">
-        <Link>roslyn\%(RecursiveDir)%(Filename)%(Extension)</Link>
-      </RoslynCompilerFiles>
+      <RoslynCompilerFiles Include="$(RoslynToolPath)\**\*" />
```

I'd suggest to leverage item transform:

```suggestion
      <CompilerFilesToInclude Include="$(RoslynToolPath)\**\*" />
      <RoslynCompilerFiles Include="@(CompilerFilesToInclude )">
        <Link>roslyn\%(RecursiveDir)%(Filename)%(Extension)</Link>
      </RoslynCompilerFiles>
```

In order not to lose the `Link` metadata.

More details: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-batching?view=vs-2022#using-helper-item-and-transform